### PR TITLE
Dashboard skeleton

### PR DESCRIPTION
**Pull request by damienaries**
### What this PR does (required):
- List the components (empty for now) that will be rendered within the logged in user dashboard. 
- Fully functional navigation, including dropdown for profile routes
- Set up empty component for sitters to register

### Screenshots / Videos (required):
- Not logged in
- Screen Shot 2021-08-12 at 3 02 28 PM
- Logged in
Screen Shot 2021-08-12 at 3 03 55 PM

### Any information needed to test this feature (required)
- Try and click on all the links and make sure they work!

### Any issues with the current functionality (optional):
- Routing to empty dashboard components still needs to be defined and added

### CODE REVIEW

File: client/src/components/Navigation/MemberLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const MemberLinks: FC = () => {
+  const classes = useStyles();
+  const history = useHistory();
+  const [role, setRole] = useState('owner');
+  const [profileOpen, setProfileOpen] = useState(false);
+
+  // Toggle profile dropdown, auto close after 10 seconds
+  const handleProfileClick = () => {
+    setProfileOpen(!profileOpen);
```
Comment (rajivtitus) [inline client/src/components/Navigation/MemberLinks.tsx:21]: If your new state update depends on the previous state, always use the functional form of setState which accepts as argument a function that returns a new state.

Ref: https://stackoverflow.com/questions/40359800/how-to-toggle-boolean-state-of-react-component

File: client/src/components/Navigation/MemberLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    setTimeout(() => {
+      setProfileOpen(false);
```
Comment (rajivtitus) [inline client/src/components/Navigation/MemberLinks.tsx:23]: 👍🏽

File: client/src/components/Navigation/MemberLinks.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  // Logout handler
+  const handleLogout = () => {
+    logout();
+  };
+
+  return (
+    <ul className={classes.memberLinks}>
+      {/* <li className={classes.memberLinkItem}>Notifications</li> */}
```
Comment (rajivtitus) [inline client/src/components/Navigation/MemberLinks.tsx:34]: Please remove commented code

File: client/src/components/Navigation/MemberLinks.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }, 10000);
+  };
+
+  // Logout handler
+  const handleLogout = () => {
+    logout();
+  };
+
+  return (
+    <ul className={classes.memberLinks}>
```
Comment (rajivtitus) [inline client/src/components/Navigation/MemberLinks.tsx:33]: Look into using Material UI's `<List>` API :)

Ref: https://material-ui.com/components/lists/

File: client/src/components/Navigation/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  // Go to home page
+  const handleClick = () => {
+    history.push('/');
+  };
+
+  return (
+    <div className={classes.nav}>
```
Comment (rajivtitus) [inline client/src/components/Navigation/Navbar.tsx:20]: Maybe look into the MUI `<AppBar>` for use here

Ref: https://material-ui.com/components/app-bar/

File: client/src/components/Navigation/PublicLinks.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import { useHistory, Link } from 'react-router-dom';
+import useStyles from './useStyles';
+import Button from '@material-ui/core/Button';
+
+const PublicLinks: FC = () => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  return (
+    <ul className={classes.publicLinks}>
```
Comment (rajivtitus) [inline client/src/components/Navigation/PublicLinks.tsx:11]: Same as comment above for `<List>`

File: client/src/components/Navigation/PublicLinks.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      <Link to="/become-a-sitter" className={classes.sitterLink}>
+        <a>Become a sitter</a>
```
Comment (rajivtitus) [inline client/src/components/Navigation/PublicLinks.tsx:13]: Maybe consider using a react-router `<Link>` instead? :)

**Review by rajivtitus — COMMENTED** [review]
Good effort! I've made some comments, please have a look and address when possible :)

File: client/src/components/Navigation/PublicLinks.tsx (hunk 7 above)
Comment (damienaries) [reply client/src/components/Navigation/PublicLinks.tsx:13]: Good question. I kind of did that thinking I remembered it from somewhere, and when it didn't give me any errors I thought I was right :) But of course I'll remove it if it's not useful

File: client/src/components/Navigation/MemberLinks.tsx (hunk 3 above)
Comment (damienaries) [reply client/src/components/Navigation/MemberLinks.tsx:34]: Got it

File: client/src/components/Navigation/MemberLinks.tsx (hunk 2 above)
Comment (damienaries) [reply client/src/components/Navigation/MemberLinks.tsx:23]: Thanks. Same as above for the setState though?

File: client/src/components/Navigation/MemberLinks.tsx (hunk 1 above)
Comment (damienaries) [reply client/src/components/Navigation/MemberLinks.tsx:21]: Like so?
setProfileOpen(prevProfileOpen=> !prevProfileOpen);

File: client/src/components/Navigation/PublicLinks.tsx (hunk 6 above)
Comment (damienaries) [reply client/src/components/Navigation/PublicLinks.tsx:11]: Received, for all three previous comments, I do need to remember to look into MUI's components.

File: client/src/components/Navigation/PublicLinks.tsx (hunk 7 above)
Comment (damienaries) [reply client/src/components/Navigation/PublicLinks.tsx:13]: Unimportant, but I just remembered where I got that from, it's the way to write them in Next.js
https://nextjs.org/docs/api-reference/next/link

File: client/src/components/Navigation/Navbar.tsx (hunk 5 above)
Comment (damienaries) [reply client/src/components/Navigation/Navbar.tsx:20]: I did, and somehow it's messing up my element positioning in the navbar, and no reacting to flexbox properties as i'm used to. Is that a mui thing? I am new with it. I thought it was pretty close as it was, but I did remove the div to make it a <nav> element.
Let me know what you think, I can keep trying to use the <AppBar />

Comment (damienaries) [reply client/src/components/Navigation/Navbar.tsx:20]: Missing some words i wrote in tags. I  changed the div into a nav element. and I said I can keep trying to use the AppBar if you prefer

Comment (damienaries) [reply client/src/components/Navigation/Navbar.tsx]: Nevermind, I found a resource that helped me. All elements are MUI now

File: client/src/components/Navigation/PublicLinks.tsx (hunk 6 above)
Comment (rajivtitus) [reply client/src/components/Navigation/PublicLinks.tsx:11]: Please change HTML element with MUI component here

File: client/src/components/Navigation/PublicLinks.tsx (hunk 7 above)
Comment (rajivtitus) [reply client/src/components/Navigation/PublicLinks.tsx:13]: Please remove hyperlink tag as we are using react-router Link for routing :)

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes.

Just a couple of minor comments, please address before merging.

**Pull request merged**